.github/workflows/tests-pro-integration.yml

```diff
@@ -310,12 +310,20 @@ jobs:
           source .venv/bin/activate
           bin/test_localstack_pro.sh
 
-      - name: Determine test selection
+      - name: Determine Test Selection
         if: ${{ env.TESTSELECTION_PYTEST_ARGS }}
         working-directory: localstack
         run: |
           . ../localstack-ext/.venv/bin/activate
-          python -m localstack.testing.testselection.scripts.generate_test_selection_from_commits $(pwd) ${{ github.event.pull_request.base.sha }} ${{ github.event.pull_request.head.sha }} target/testselection/test-selection.txt
+          if [ -z "${{ github.event.pull_request.base.sha }}" ]; then
+            python -m localstack.testing.testselection.scripts.generate_test_selection_from_pr $(pwd) target/testselection/test-selection.txt || (mkdir -p target/testselection && echo "SENTINEL_ALL_TESTS" >> target/testselection/test-selection.txt)
```

Comment:
Then we're in a bit of a pickle. To have this working on `master` you'd need to allow selecting base and head commits manually.

But I don't really see a use case of running it on `master`. If there is a specific one we'd like to have I can look into how we can support that.

Reply:
I do see a reason why to run "this" (meaning the pipeline) on master. I do that regularly, and it failed last time. The test selection is enabled by default, also for `workflow_dispatch` events.
This could easily be fixed by just having a gracious fallback to "run all tests" if the detection fails.

Reply:
The gracious fallback is added in the line we're commenting on.

```|| (mkdir -p target/testselection && echo "SENTINEL_ALL_TESTS" >> target/testselection/test-selection.txt)```

Do you have something different in mind here?

Reply:
Well that was my question in the first place. If it works just fine when being executed on master with the default settings, then it's okay. 😜